drivers/src/sha256.rs

```diff
@@ -215,9 +244,6 @@ impl Sha256 {
             sha256.ctrl().write(|w| w.mode(true).init(false).next(true));
         }
```

Comment:
Why is this removed?

Reply:
Since this change is unrelated to the ROM integrity testing, can we make this as a separate change?

Reply:
For a multi-block case, how do we ensure that the crypto engine is not processing the data in the block registers? digest_block simply copies the data to the block registers and then calls digest_op, which then checks for status.ready. Previously, digest_op would have only returned when the processing was complete.

Reply:
I asked Michael:

> Kor: I'm trying to stream data to the sha256 peripheral as fast as possible, do I need to wait for the ready bit before writing to the SHA256_BLOCK register, or do I only have to check that before writing the command? (edited) 
> 
> Michael: I believe init/next commands will latch the data in BLOCK. You can begin writing the next block while the command is running setting next again only after ready
> 
> Kor: ok, that's my interpretation of the RTL as well
> 
> Michael: That's how I do it for sha accelerator and any tests we've written. I only check ready prior to starting the next command

You can see it grabbing the entire register immediately after the command is received:

https://github.com/chipsalliance/caliptra-rtl/blob/main/src/sha256/rtl/sha256_w_mem.v#L209

Reply:
Is this really needed? Can we continue to use the old interface. How much does this improve the time by?

Reply:
digest_blocks_raw() is definitely required, as it is undefined behavior to create a slice reference to address zero. No way around that one.

Removing the wait for status::valid saves 45,483 cycles, due to the additional pipelining.

Reply:
Is ~45000 a big win that warrents such large changes this late?

Reply:
I honestly don't know, since we don't have much guidance on performance. This does delay how long it takes the ROM to be ready for the first measurements.

I did the optimization because I wanted to collect perf numbers with the optimal implementation when I wrote this 2 months ago.

If you're not comfortable, I can revert.

Reply:
Ok, in the interest of getting this PR in sooner, I've reverted the validity check optimization.